File: Java/ByTier/A/2128_A_Recycling_Center.java
// Problem: CF 2128 A - Recycling Center
// https://codeforces.com/contest/2128/problem/A

/*
 * Problem: Recycling Center
 * 
 * Purpose:
 *   Given n trash bags with weights, we must destroy all of them. Each second,
 *   we choose one bag to destroy (costing 1 coin if weight > c, 0 otherwise),
 *   then all remaining bags' weights are doubled.
 *   
 *   Goal: Find the minimum number of coins needed to destroy all bags.
 *   
 * Algorithm:
 *   Greedy approach:
 *     - Sort bags in ascending order.
 *     - Process from largest to smallest (reverse iteration).
 *     - For each bag, if its current weight (after doubling) is <= c, we can
 *       destroy it for free; otherwise, we must pay 1 coin.
 *     - We maintain a counter 'doube' which represents how many times the bag's
 *       weight was doubled before processing it.
 * 
 * Time Complexity: O(n log n) per test case due to sorting.
 * Space Complexity: O(n) for the array storage.
 */

import java.util.*;
public class Main {
    public static void main(String[] args) {
       Scanner sc=new Scanner(System.in);
       int t=sc.nextInt();
       while(t-->0)
       {
         int n=sc.nextInt();        // Read number of bags
         int c=sc.nextInt();        // Read threshold weight
         int arr[]=new int[n];      // Array to store weights
         for(int i=0;i<n;i++)
         {
           arr[i]=sc.nextInt();     // Read each bag's weight
         }
         Arrays.sort(arr);          // Sort in ascending order
         int doube=0;               // Counter for number of doublings
         for(int i=n-1;i>=0;i--)    // Process from largest to smallest
         {
           if(arr[i]*(1<<doube)<=c) // Check if bag can be destroyed for free
           {
             doube++;               // Increment doubling count
           }
         }
         System.out.println(n-doube); // Print coins spent = total bags - free bags
       }
  }
}


// https://github.com/VaHiX/CodeForces/

File: Kotlin/ByTier/A/2128_A_Recycling_Center.kt
// Problem: CF 2128 A - Recycling Center
// https://codeforces.com/contest/2128/problem/A

/**
 * Problem: Recycling Center
 * 
 * Description:
 * We have n trash bags with weights a[i]. Each second, we choose one bag to destroy:
 * - If weight > c, it costs 1 coin.
 * - After destruction, all remaining bags double in weight.
 * Goal: Minimize total coins spent to destroy all bags.
 * 
 * Algorithm:
 * Greedy strategy with binary search on answer.
 * Sort the array. For a given number of bags (m) that we will destroy,
 * check if it's possible to destroy them using the greedy method:
 * - Always destroy the heaviest available bag (greedy choice).
 * - If the weight of current bag is <= c, no coin is paid.
 * 
 * Time Complexity: O(n log n + T * n * log n), where T = number of test cases
 * Space Complexity: O(n)
 */
import java.io.BufferedReader
import java.io.InputStreamReader
import java.io.PrintWriter
import java.io.StreamTokenizer
import java.util.*
private val sc = Scanner( System.`in` )
private val br = BufferedReader( InputStreamReader( System.`in` ) )
private val st = StreamTokenizer( br )
private val pw = PrintWriter( System.out )
fun main() {
    solve()
    pw.close()
}
private fun solve() {
    val tn = st.nextInt()
    for ( test in 1 .. tn ) {
        solveTest()
    }
}
private fun solveTest() {
    val n = st.nextInt()
    val c = st.nextInt()
    val a = LongArray( n ) { st.nextInt().toLong() }
    a.sort()
    var l = 0
    var r = n
    while ( l < r ) {
        val m = ( l + r + 1 ) / 2 // Binary search on number of bags to destroy
        var ok = true
        for ( i in 0 until m ) {
            // Destroy the heaviest bag among the remaining ones (greedy)
            if ( a[m - i - 1] shl i > c ) { // Check if cost is > c (shl i means multiply by 2^i)
                ok = false
                break
            }
        }
        if ( ok ) {
            l = m // Can destroy m bags, try to increase
        } else {
            r = m - 1 // Cannot destroy m bags, decrease
        }
    }
    pw.println( n - l ) // Remaining bags that don't get destroyed = n - l
}
private fun StreamTokenizer.nextInt(): Int {
    nextToken()
    return nval.toInt()
}


// https://github.com/VaHiX/CodeForces/